ocp_utilities/infra.py

```diff
@@ -81,6 +84,46 @@ def assert_pods_failed_or_pending(pods):
         )
 
 
+def assert_nodes_in_healthy_condition(nodes):
+    """
+    Validates nodes are in a healthy condition.
+    Nodes Ready condition is True and none of the following node conditions is True:
+        - DiskPressure
+        - MemoryPressure
+        - PIDPressure
+        - NetworkUnavailable
+
+    Args:
+         nodes(list): List of Node objects
+
+    Raises:
+        NodesNotHealthyConditionError: if any nodes DiskPressure
+        MemoryPressure, PIDPressure, NetworkUnavailable condition is True
+    """
+    LOGGER.info("Verify all nodes are in a healthy condition")
+    unhealthy_nodes_with_conditions = []
```

Comment:
Could we use a dictionary over a list of lists? I feel using a dictionary just would make things cleaner. Within the dictionary there could be a key per node and the value is a list of the condition types that are not healthy.

Pseudocode (not tested)

```python
import json

unhealthy_nodes = {}
for node in nodes:
    unhealthy_nodes[node.name] = []
    for condition in node.instance.status.conditions:
        if (
            condition.type != Node.Condition.READY
            and condition.status == Node.Condition.Status.TRUE
        ):
            unhealthy_nodes[node.name].append(condition.type)

    if len(unhealthy_nodes[node.name] == 0):
        unhealthy_nodes.pop(node.name)

if len(unhealthy_nodes):
    raise NodesNotHealthyConditionError(
        f"Following are nodes with unhealthy conditions: {json.dumps(unhealthy_nodes)}"
    )

```

Reply:
@rnetser what's the guidance here?

Reply:
I must agree, more readable and fit the purpose.

Reply:
agree